[BUG] HTTP 500 on DELETE /api/documents/{id}/notes/ with missing or invalid note id

### Description

Sending a `DELETE` request to `/api/documents/{id}/notes/` with a missing,  empty, or non-existent `id` query parameter causes an unhandled exception and returns HTTP 500 instead of a proper error response.

I believe the API should return a HTTP 400 instead, as this is clearly a client problem and not a server one.
Noticed this behavior while playing with the notes through API. I don't think this behavior can arise from the UI. Although it would be nice to enforce this check on the API side.


### Steps to reproduce

To reproduce locally:

```
# Try to delete a non-existing note_id -> HTTP 500
curl -X DELETE -w %{http_code} 'http://localhost:8000/api/documents/1/notes/?id=10' -H "Authorization: Token <REDACTED>"

# Try to delete a note while providing an empty id parameter -> HTTP 500
curl -X DELETE -w %{http_code} 'http://localhost:8000/api/documents/1/notes/?id=' -H "Authorization: Token <REDACTED>"

# Try to delete a note without providing an id parameter -> HTTP 500
curl -X DELETE -w %{http_code} 'http://localhost:8000/api/documents/1/notes/' -H "Authorization: Token <REDACTED>"
```


### Webserver logs

```bash
Internal Server Error: /api/documents/1/notes/
Traceback (most recent call last):
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/django/views/decorators/csrf.py", line 65, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/rest_framework/viewsets.py", line 125, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/rest_framework/views.py", line 515, in dispatch
    response = self.handle_exception(exc)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/rest_framework/views.py", line 475, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/rest_framework/views.py", line 486, in raise_uncaught_exception
    raise exc
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/rest_framework/views.py", line 512, in dispatch
    response = handler(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/src/documents/views.py", line 1505, in notes
    note = Note.objects.get(id=int(request.GET.get("id")), document=doc)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/django/db/models/query.py", line 635, in get
    raise self.model.DoesNotExist(
documents.models.Note.DoesNotExist: Note matching query does not exist.
[2026-04-15 17:32:08,044] [ERROR] [django.request] Internal Server Error: /api/documents/1/notes/
Traceback (most recent call last):
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/django/views/decorators/csrf.py", line 65, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/rest_framework/viewsets.py", line 125, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/rest_framework/views.py", line 515, in dispatch
    response = self.handle_exception(exc)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/rest_framework/views.py", line 475, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/rest_framework/views.py", line 486, in raise_uncaught_exception
    raise exc
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/rest_framework/views.py", line 512, in dispatch
    response = handler(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/src/documents/views.py", line 1505, in notes
    note = Note.objects.get(id=int(request.GET.get("id")), document=doc)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ubuntu/paperless-ngx/.venv/lib/python3.12/site-packages/django/db/models/query.py", line 635, in get
    raise self.model.DoesNotExist(
documents.models.Note.DoesNotExist: Note matching query does not exist.
[15/Apr/2026 17:32:08] "DELETE /api/documents/1/notes/?id=10 HTTP/1.1" 500 142494
```

### Browser logs

```bash

```

### Services logs

```bash

```

### Paperless-ngx version

 2.20.14

### Host OS

Ubuntu 24.04

### Installation method

Docker - official image

### System status

```json
{
    "pngx_version": "2.20.14",
    "server_os": "Linux-6.8.0-107-generic-x86_64-with-glibc2.41",
    "install_type": "docker",
    "storage": {
        "total": 50884108288,
        "available": 39117668352
    },
    "database": {
        "type": "postgresql",
        "url": "paperless",
        "status": "OK",
        "error": null,
        "migration_status": {
            "latest_migration": "paperless_mail.0001_initial_squashed_0009_mailrule_assign_tags",
            "unapplied_migrations": []
        }
    },
    "tasks": {
        "redis_url": "redis://broker:6379",
        "redis_status": "OK",
        "redis_error": null,
        "celery_status": "OK",
        "celery_url": "celery@b32c3a50801c",
        "celery_error": null,
        "index_status": "OK",
        "index_last_modified": "2026-04-15T16:21:53.266823Z",
        "index_error": null,
        "classifier_status": "OK",
        "classifier_last_trained": "2026-04-15T17:05:07.258788Z",
        "classifier_error": null,
        "sanity_check_status": "OK",
        "sanity_check_last_run": "2026-04-14T12:13:42.107818Z",
        "sanity_check_error": null
    },
    "websocket_connected": "ERROR"
}
```

### Browser

Firefox

### Configuration changes

```
PAPERLESS_URL=http://192.168.1.21:8000
PAPERLESS_CSRF_TRUSTED_ORIGINS=https://<REDACTED>,http://192.168.1.21:8000
PAPERLESS_ALLOWED_HOSTS=https://<REDACTED>,192.168.1.21
PAPERLESS_CORS_ALLOWED_HOSTS=https://<REDACTED>,http://192.168.1.21:8000

PAPERLESS_OCR_CLEAN=clean
PAPERLESS_OCR_DESKEW=true
PAPERLESS_OCR_ROTATE_PAGES=true
PAPERLESS_OCR_LANGUAGE=fra
PAPERLESS_OCR_USER_ARGS={"continue_on_soft_render_error": true}
PAPERLESS_OCR_MODE=skip
PAPERLESS_OCR_SKIP_ARCHIVE_FILE=with_text
PAPERLESS_OCR_OUTPUT_TYPE=pdf

PAPERLESS_CONSUMER_POLLING=86400
# PAPERLESS_CONSUMER_DELETE_DUPLICATES=true
PAPERLESS_LOGROTATE_MAX_BACKUPS=10

PAPERLESS_CONSUMER_RECURSIVE=true
PAPERLESS_CONSUMER_SUBDIRS_AS_TAGS=true

PAPERLESS_THREADS_PER_WORKER=1
PAPERLESS_TASK_WORKERS=1

PAPERLESS_SECRET_KEY=<REDACTED>
PAPERLESS_FORCE_SCRIPT_NAME=<REDACTED>

PAPERLESS_REDIS=redis://broker:6379
PAPERLESS_DBHOST=db
```

### Please confirm the following

- [x] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [x] This issue is not about the OCR or archive creation of a specific file(s). Otherwise, please see above regarding OCR tools.
- [x] I have already searched for relevant existing issues and discussions before opening this report.
- [x] I have updated the title field above with a concise description.

--- a/src/documents/tests/test_api_documents.py
+++ b/src/documents/tests/test_api_documents.py
@@ -3146,6 +3146,56 @@ class TestDocumentApi(DirectoriesMixin, DocumentConsumeDelayMixin, APITestCase):
         # modified was updated to today
         self.assertEqual(doc.modified.day, timezone.now().day)
 
+    def test_delete_note_missing_id(self) -> None:
+        """
+        GIVEN:
+            - Existing document
+        WHEN:
+            - API DELETE request to notes endpoint without an id query param
+            - API DELETE request to notes endpoint with an empty id query param
+        THEN:
+            - HTTP 400 is returned
+        """
+        doc = Document.objects.create(
+            title="test",
+            mime_type="application/pdf",
+            content="this is a document",
+        )
+
+        response = self.client.delete(
+            f"/api/documents/{doc.pk}/notes/",
+            format="json",
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        response = self.client.delete(
+            f"/api/documents/{doc.pk}/notes/?id=",
+            format="json",
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_delete_note_nonexistent_id(self) -> None:
+        """
+        GIVEN:
+            - Existing document, no notes
+        WHEN:
+            - API DELETE request to notes endpoint with a non-existent note id
+        THEN:
+            - HTTP 404 is returned
+        """
+        doc = Document.objects.create(
+            title="test",
+            mime_type="application/pdf",
+            content="this is a document",
+        )
+
+        response = self.client.delete(
+            f"/api/documents/{doc.pk}/notes/?id=99999",
+            format="json",
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+
     def test_get_notes_no_doc(self) -> None:
         """
         GIVEN:

--- a/src/documents/tests/test_api_documents.py
+++ b/src/documents/tests/test_api_documents.py
@@ -3174,6 +3174,27 @@ class TestDocumentApi(DirectoriesMixin, DocumentConsumeDelayMixin, APITestCase):
         )
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
 
+    def test_delete_note_invalid_id(self) -> None:
+        """
+        GIVEN:
+            - Existing document
+        WHEN:
+            - API DELETE request to notes endpoint with a non-integer note id
+        THEN:
+            - HTTP 400 is returned
+        """
+        doc = Document.objects.create(
+            title="test",
+            mime_type="application/pdf",
+            content="this is a document",
+        )
+
+        response = self.client.delete(
+            f"/api/documents/{doc.pk}/notes/?id=notaninteger",
+            format="json",
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
     def test_delete_note_nonexistent_id(self) -> None:
         """
         GIVEN:

--- a/src/documents/views.py
+++ b/src/documents/views.py
@@ -1502,7 +1502,13 @@ class DocumentViewSet(
             ):
                 return HttpResponseForbidden("Insufficient permissions to delete notes")
 
-            note = Note.objects.get(id=int(request.GET.get("id")), document=doc)
+            note_id = request.GET.get("id")
+            if not note_id:
+                raise ValidationError({"id": "This field is required."})
+            try:
+                note = Note.objects.get(id=int(note_id), document=doc)
+            except (Note.DoesNotExist, ValueError):
+                raise NotFound(f"Note {note_id} does not exist.")
             if settings.AUDIT_LOG_ENABLED:
                 LogEntry.objects.log_create(
                     instance=doc,

--- a/src/documents/views.py
+++ b/src/documents/views.py
@@ -1506,9 +1506,10 @@ class DocumentViewSet(
             if not note_id:
                 raise ValidationError({"id": "This field is required."})
             try:
-                note = Note.objects.get(id=int(note_id), document=doc)
-            except (Note.DoesNotExist, ValueError):
-                raise NotFound(f"Note {note_id} does not exist.")
+                note_id_int = int(note_id)
+            except ValueError:
+                raise ValidationError({"id": "A valid integer is required."})
+            note = get_object_or_404(Note, id=note_id_int, document=doc)
             if settings.AUDIT_LOG_ENABLED:
                 LogEntry.objects.log_create(
                     instance=doc,

